lib/ansible/modules/files/blockfromfile.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Evan Kaufman <evan@digitalflophouse.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+---
+module: blockfromfile
+author:
+  - Evan Kaufman (@EvanK)
+version_added: "2.5"
+short_description: Search file from remote node using a provided regular expression
+description:
+  - This module will search a remote file for all instances of a pattern.
+    Effectively the inverse of M(replace).
+options:
+  src:
+    required: true
+    aliases: [ name, srcfile ]
+    description:
+      - The file to search.
+  regexp:
+    required: true
+    description:
+      - The regular expression to look for in the contents of the file.
+        Uses Python regular expressions; see
+        U(http://docs.python.org/2/library/re.html).
+        Uses multiline mode, which means C(^) and C($) match the beginning
+        and end respectively of I(each line) of the file.
+  fail_on_missing:
+    default: false
+    description:
+      - Makes it fails when the source file is missing.
+notes:
+   - "See also: M(replace)"
+"""
+
+EXAMPLES = r"""
+# finds one named group, "priority"
+- blockfromfile:
+    src: /etc/keepalived/keepalived.conf
+    regexp: '^[ \t\f\v]*priority[ \t\f\v]*(?P<priority>\d+)[ \t\f\v]*'
+
+# finds two named groups, "address" and "hostnames"
+- blockfromfile:
+    src: /etc/hosts
+    regexp: '^[ \t\f\v]*(?P<address>[a-f\d.:]+)[ \t\f\v]*(?P<hostnames>(?:\S+[ \t\f\v]*)+)'
+
+# finds one unnamed group
+- blockfromfile:
+    src: /etc/sudoers
+    regexp: '^[ \t\f\v]*(\S+)(?:[ \t\f\v]+\S*)+NOPASSWD:ALL'
+"""
+
+RETURN = """
+matches:
+    description: numbered or named regular expression groups
+    returned: changed
+    type: list
+    sample: [ { "groups": [ "VALUE" ], "named_groups": { "NAME": "VALUE" } } ]
+"""
+
+import re
+import os
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.pycompat24 import get_exception
+from ansible.module_utils.basic import AnsibleModule
+
+
+def find_from_content(regexp, content):
+    result = []
+    found = re.finditer(regexp, content, re.MULTILINE)
+
+    for match in found:
+        result.append({
+            'groups': match.groups(),
+            'named_groups': match.groupdict(),
+        })
+
+    return result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            src=dict(required=True, aliases=['name', 'srcfile', 'path'], type='path'),
```

Comment:
I always advice to start with the type, and always add a type (even if it is a str, and str is the default type).

Reply:
Also `path` is usually the default parameter if there's no destionation (e.g. **stat** module or **file** module).
And `srcfile` is so unusual (I haven't seen it before) so I would leave it out altogether. Same for `name`.
There's no advantage of having more aliases, to the contrary. As a courtesy to making it easier for end-users, we tend to have `src` and `path` both.

Reply:
And I would use `path` in the examples as well.